src/modules/fancyzones/editor/FancyZonesEditor/Models/LayoutModel.cs

```diff
@@ -219,30 +221,65 @@ public static ObservableCollection<LayoutModel> LoadCustomModels()
                     var info = current.GetProperty(InfoJsonTag);
                     if (type.Equals(GridJsonTag))
                     {
+                        bool error = false;
+
                         int rows = info.GetProperty(RowsJsonTag).GetInt32();
                         int columns = info.GetProperty(ColumnsJsonTag).GetInt32();
 
                         List<int> rowsPercentage = new List<int>(rows);
                         JsonElement.ArrayEnumerator rowsPercentageEnumerator = info.GetProperty(RowsPercentageJsonTag).EnumerateArray();
+
+                        List<int> columnsPercentage = new List<int>(columns);
+                        JsonElement.ArrayEnumerator columnsPercentageEnumerator = info.GetProperty(ColumnsPercentageJsonTag).EnumerateArray();
+
+                        if (rows <= 0 || columns <= 0 || rowsPercentageEnumerator.Count() != rows || columnsPercentageEnumerator.Count() != columns)
+                        {
+                            error = true;
+                        }
+
                         while (rowsPercentageEnumerator.MoveNext())
                         {
-                            rowsPercentage.Add(rowsPercentageEnumerator.Current.GetInt32());
+                            int percentage = rowsPercentageEnumerator.Current.GetInt32();
+                            if (error || percentage <= 0)
+                            {
+                                error = true;
+                                break;

```

Comment:
In my opinion `while (... && !error)` looks more clear, but it's optional change

Reply:
You're right. Rushing things never helps :) Thanks! Fixed